store/modules/data/actions.js

```diff
@@ -43,15 +43,28 @@ export default {
           state.stim_waveforms[well_idx].x_data_points.push(x);
           state.stim_waveforms[well_idx].y_data_points.push(101000);
 
-          next_x
-            ? state.stim_fill_assignments[well_idx].push([
-                protocol_flags[idx],
-                [
-                  [x, 101000],
-                  [next_x, 101000],
-                ],
-              ])
-            : (state.last_protocol_flag[well_idx] = [x, protocol_flags[idx]]);
+          if (next_x) {
+            state.stim_fill_assignments[well_idx].push([
+              protocol_flags[idx],
+              [
+                [x, 101000],
+                [next_x, 101000],
+              ],
+            ]);
+          } else if (!next_x && new_well_values.length == 1) {
```

Comment:
Can just be `new_well_values.length == 1`

Reply:
true